app/controllers/api/offers_controller.rb

```diff
@@ -2,26 +2,62 @@
 
 class Api::OffersController < ApplicationController
   before_action :authenticate_user!
+  rescue_from ActiveRecord::RecordNotFound, with: :render_offer_error
 
   def create
-    offer = Offer.create(check_and_convert_parameters)
+    offer = current_user.offers.create(offer_parameters)
     if offer.persisted?
       render json: { message: 'Your offer has been sent!' }
     else
-      render json: { message: offer.errors.full_messages.join(". ") }, status: 422
+      render json: { message: offer.errors.full_messages.join('. ') }, status: 422
     end
   rescue StandardError => e
     render json: { message: e.message }, status: 422
   end
 
+  def update
+    offer = Offer.find(params[:id])
+    request = Request.find(offer.request.id)
+
+    case params[:activity]
+    when 'accepted'
+      offer.update(status: params[:activity])
+      request.update(helper: offer.helper)
+      render json: { offer: offer.id, message: "offer is #{offer.status}" }
+    when 'declined'
+      offer.update(status: params[:activity])
+      render json: { offer: offer.id, message: "offer is #{offer.status}" }
```

Comment:
I would like the message to say 'You accepted/declined help from helper@mail.com', like in the Lo-Fi, but that can be done later.

Reply:
Good, vill change!